### PR TITLE
Adds current-or-terminal-occupied-room to dtos

### PR DESCRIPTION
**Pull request by pcsouza**
@eldavido PTAL

This adds information onto summary and complete reservation DTOs about the room occupancy status.

My plan is that the occupied room will be stored in materialized state as a room number string and an indicator (guid in the database) on the reservation model object. I'm calling this current-or-terminal occupancy because we have mutually-exclusive cases where we want to know the room occupancy of the reservation:

1) Upon observing a checked-in reservation, we wish to know the room it currently occupies
2) Upon observing a checked-out reservation, we wish to know the last room it occupied

These two cases occur in mutual exclusivity because occupancy is hard-released on checkout. Therefore, I decided to merge them into one field because,

1) It's simpler just to leave the field untouched when the reservation checks out
2) It's simpler for consumers, since they don't need to be concerned about multiple room occupancy fields based on the reservation state.

### CODE REVIEW

File: proto/booking/reservations/complete_open_reservation.proto [hunk 1]
```diff
@@ -30,4 +31,7 @@ message CompleteOpenReservation {
 	holms.types.primitive.MonetaryAmount unpaid_guarantee_balance = 13;
 	holms.types.operations.housekeeping.HousekeepingTime hk_time_preference = 15;
 	string vehicle_plate_information = 16;
+
+	string current_or_terminal_occupied_room_number = 17;
```
Comment (eldavido) [inline proto/booking/reservations/complete_open_reservation.proto]: maybe just `last_occupied_room_number`?

File: proto/booking/reservations/complete_open_reservation.proto [hunk 2]
```diff
@@ -30,4 +31,7 @@ message CompleteOpenReservation {
...
+	holms.types.operations.rooms.RoomIndicator current_or_terminal_occupied_room = 18;
```
Comment (eldavido) [inline proto/booking/reservations/complete_open_reservation.proto]: again, just `last_occupied_room`?

File: proto/booking/reservations/reservation_summary.proto [hunk 3]
```diff
@@ -25,4 +27,7 @@ message ReservationSummary {
 	.holms.types.tenancy_config.indicators.PropertyIndicator property = 9;
 	string group_booking_id = 10;
 	string vehicle_plate_information = 11;
+
+	string current_or_terminal_occupied_room_number = 12;
```
Comment (eldavido) [inline proto/booking/reservations/reservation_summary.proto]: `last_occupied_room_number`

File: proto/booking/reservations/reservation_summary.proto [hunk 4]
```diff
@@ -25,4 +27,7 @@ message ReservationSummary {
...
+	holms.types.operations.rooms.RoomIndicator current_or_terminal_occupied_room = 13;
```
Comment (eldavido) [inline proto/booking/reservations/reservation_summary.proto]: `last_occupied_room`

**Comment by pcsouza** [comment]
Per discussion will be splitting this into separate fields: current and terminal

Current: Is populated with the most-recent room occupation event's information. Becomes de-populated when the reservation de-occupies a room without re-assignment (typically check-out, but could occur in musical-chairs house ops scenarios as well)

Terminal follows the same population rules as current, but never de-populates.

Current is for knowing the immediate state of the reservation's occupancy while it's checked in.
Terminal is for knowing the last room the reservation was in, regardless of its checked-in state, or ongoing house reorganization efforts.

**Pull request merged**